src/python/PyBindImath/PyImathVec.cpp

```diff
@@ -0,0 +1,17 @@
+//
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright Contributors to the OpenEXR Project.
+//
+
+#include "PyImath.h"
+#include <ImathVec.h>
+
+namespace PyImath {
+
+void register_imath_vec(pybind11::module& m)
+{
+    // TODO fill in
+    pybind11::class_<Imath::V3f>(m, "V3f");
```

Comment:
It would be great to implement the Python bindings so that there is a class hierarchy matching C++. So have a `Vec3` class, and the type-specific variations inheriting from that. This would improve type hinting and checking to support things like `isinstance(vector, Vec3)` and `def my_function(vector: Vec3) -> Matrix44:`. As an example of this approach, OCIO has implemented `Transform` and its subclasses in a similar way: 
https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/main/src/bindings/python/PyTransform.cpp

Reply:
I'm happy to collaborate on this work if help is needed. I was getting ready to start experimenting myself with a pybind11 implementation of the bindings when I noticed this PR. Awesome to see this work is already started :)

Reply:
Thanks, @michdolan, we're happy to have the help! And the class hierarchy sounds like a nice structure.

@lgritz, you've requested changes, are you OK with this as is for now?

I still think a separate `pybind11` subdirectory in parallel with `python` would make sense, since that will make room for the pybind11 tests, and also a fresh,  boost-free, pybind-specific config, but it's fine to deal with that later.

Reply:
Yeah, I think separating it out in a parallel directory should be done later.  I'm currently planning on creating python tests that should pass with both the boost bindings and the pybind11 ones, and it will be more convenient for the time being to be under python.